search/ui/class-vip-search-widget.php

```diff
@@ -0,0 +1,454 @@
+		}
+	}
+
+	/**
+	 * Enqueue scripts and styles for the frontend.
+	 *
+	 */
+	public function enqueue_frontend_scripts() {
+		if ( ! is_active_widget( false, false, $this->id_base, true ) ) {
+			return;
+		}
+
+		wp_enqueue_script(
+			'vip-search-widget',
+			plugins_url( 'js/search-widget.js', __FILE__ ),
+			array(),
+			1.0,
+			true
+		);
+
+		wp_enqueue_style( 'vip-search-widget', plugins_url( 'css/search-widget-frontend.css', __FILE__ ), array(), 1.0 );
+	}
+
+	/**
+	 * Get the list of valid sort types/orders.
+	 *
+	 * @return array The sort orders.
+	 *
+	 */
+	private function get_sort_types():array {
+		return array(
+			'relevance|DESC' => is_admin() ? esc_html__( 'Relevance (recommended)', 'jetpack' ) : esc_html__( 'Relevance', 'jetpack' ),
+			'date|DESC'      => esc_html__( 'Newest first', 'jetpack' ),
+			'date|ASC'       => esc_html__( 'Oldest first', 'jetpack' ),
+		);
+	}
+
+	public function vip_search_populate_defaults( $instance ) {
+		$instance = wp_parse_args(
+			(array) $instance, array(
+				'title'              => '',
+				'search_box_enabled' => true,
+				'user_sort_enabled'  => true,
+				'sort'               => self::DEFAULT_SORT,
+				'post_types'         => array(),
+			)
+		);
+
+		return $instance;
+	}
+
+	/**
+	 * Responsible for rendering the widget on the frontend.
+	 *
+	 * @param array $args     Widgets args supplied by the theme.
+	 * @param array $instance The current widget instance.
+	 *
+	 */
+	public function widget( $args, $instance ) {
+		$instance = $this->vip_search_populate_defaults( $instance );
+
+		if ( empty( $instance['search_box_enabled'] ) && empty( $instance['user_sort_enabled'] ) ) {
+			return;
+		}
+
+		$title = ! empty( $instance['title'] ) ? $instance['title'] : '';
+
+		/** This filter is documented in core/src/wp-includes/default-widgets.php */
+		$title = apply_filters( 'widget_title', $title, $instance, $this->id_base );
+
+		echo $args['before_widget']; //phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		?>
+			<div id="<?php echo esc_attr( $this->id ); ?>-wrapper" >
+		<?php
+
+		if ( ! empty( $title ) ) {
+			$this->render_widget_title( $title, $args['before_title'], $args['after_title'] );
+		}
+
+		$default_sort            = $instance['sort'] ?? self::DEFAULT_SORT;
+		list( $orderby, $order ) = $this->sorting_to_wp_query_param( $default_sort );
+		$current_sort            = "{$orderby}|{$order}";
+
+		// we need to dynamically inject the sort field into the search box when the search box is enabled, and display
+		// it separately when it's not.
+		if ( ! empty( $instance['search_box_enabled'] ) ) {
+			$this->render_widget_search_form( $instance['post_types'], $orderby, $order );
+		}
+
+		if ( ! empty( $instance['search_box_enabled'] ) && ! empty( $instance['user_sort_enabled'] ) ) :
+			?>
+			<div class="vip-search-sort-wrapper">
+				<label>
+					<?php esc_html_e( 'Sort by', 'jetpack' ); ?>
```

Comment:
I know we technically don't have a separate text domain, but seems we should fix this copy-pasta
```suggestion
					<?php esc_html_e( 'Sort by', 'vip-search' ); ?>
```